lib/templates/plugin.js

```diff
@@ -104,6 +104,10 @@ export default (ctx, inject) => {
     const ApolloSSR = require('vue-apollo/ssr')
     beforeNuxtRender(({ nuxtState }) => {
       nuxtState.apollo = ApolloSSR.getStates(apolloProvider)
+      // Clear apollo client cache after each request
+      // Issues: https://github.com/nuxt-community/apollo-module/issues/273
+      //         https://github.com/nuxt-community/apollo-module/issues/251
+      Object.values(apolloProvider.clients).forEach(client => client.cache.reset())
```

Comment:
Wouldn't it be better to configure InMemoryCache to not cache in the first place?

Reply:
Yes, that can be a better option. So ideally we will assign cache object only in case of `process.client` at https://github.com/nuxt-community/apollo-module/blob/a8882162edbeaed9bdf8ca4db44b3d24e5a37f1d/lib/templates/plugin.js#L59

Reply:
I tried to put `<%= key %>ClientConfig.cache = <%= key %>Cache` in `if-else`, but it didn't work. Seems apollo has `InMemoryCache` enabled by default.

Reply:
InMemoryCache is default created by `vue-cli-plugin-apollo`. And it's being used in some part of the package.

https://github.com/Akryum/vue-cli-plugin-apollo/blob/a52696165732381787dafe6b8e694d4b30af4826/graphql-client/src/index.js#L59